pkg/cli/admin/catalog/downloader.go

```diff
@@ -0,0 +1,69 @@
+package catalog
+
+import (
+	"io/ioutil"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/klog"
+
+	"github.com/operator-framework/operator-registry/pkg/apprclient"
+	"github.com/operator-framework/operator-registry/pkg/appregistry"
+)
+
+// NewDownloader is a constructor for the Downloader interface
+func NewDownloader(client apprclient.Client) Downloader {
+	return &downloader{
+		client: client,
+	}
+}
+
+// Downloader is an interface that is implemented by structs that
+// implement the DownloadManifests method.
+type Downloader interface {
+	// DownloadManifests downloads the manifests in a namespace into a local directory
+	DownloadManifests(directory, namespace string) error
+	// DownloadManifests downloads manifests and returns a reference to the tmp directory that contains the manifests
+	DownloadManifestsTmp(namespace string) (directory string, err error)
```

Comment:
You're only using one of these two, drop the `DownloadManifestTmp` and only use `DownloadManifest` it's the caller choice about the directory how to create it, clean it and what else.